src/Integration.UnitTests/Suppression/SuppressedIssuesProviderTests.cs

```diff
@@ -65,17 +66,38 @@ public void Ctor_NullActiveSolutionBoundTracker_ArgumentNullException()
         }
 
         [TestMethod]
-        public void GetSuppressedIssues_NoSolutionBinding_EmptyList()
+        public void Ctor_NullSolutionBinding_ArgumentNullException()
         {
-            var actual = testSubject.GetSuppressedIssues("project guid", "file path");
+            activeSolutionBoundTracker.CurrentConfiguration = null;
+            Action act = () => new SuppressedIssuesProvider(activeSolutionBoundTracker, createProviderFunc.Object);
 
-            actual.Should().BeEmpty();
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("CurrentConfiguration");
+        }
+
+        [TestMethod]
+        public void Ctor_SolutionBindingIsStandalone_SonarQubeIssuesProviderNotCreated()
+        {
+            activeSolutionBoundTracker.CurrentConfiguration = BindingConfiguration.Standalone;
+            new SuppressedIssuesProvider(activeSolutionBoundTracker, createProviderFunc.Object);
+
+            createProviderFunc.VerifyNoOtherCalls();
```

Comment:
Presumably we're not expecting the _SuppressionsEventRaised_ event to be raised in this case?

Reply:
> The code looks good, although I'd probably just remove the check that is causing the code smell.
> 
> Is there a reason for not checking for the event being raised/not raised in the new tests?

Since it's part of the ctor, I was unable to test it -- since I can't register to the event before the class is created. Which means that AnalysisConfigMonitor can also miss it, similar to the race condition with ActiveSolutionBoundTracker -- it's possible that by the time we create SuppressedIssuesProvider, the binding event has already been raised, so we raise in CTOR, but AnalysisConfigMonitor hasn't registered to SuppressionsUpdateRequested event yet, and therefore the analysis is not triggered.
In reality this has not been an issue, since the blocking mechanism will save us (now that there is an initialization call in the CTOR).
So SuppressionsUpdateRequested event will only be raised when the binding is actively updated.

In hindsight, I probably shouldn't have gone with an event here, the whole flow is complicated enough as-is.

Reply:
Not to worry - more opportunities for refactoring later ;-)